crates/ark/src/lsp/diagnostics.rs

```diff
@@ -785,9 +785,24 @@ fn recurse_call_arguments_default(
     // TODO: Can we better handle NSE in things like `quote()` and
     // `dplyr::mutate()` so we don't have to turn off certain diagnostics when
     // we are inside a call's arguments?
-    let mut context = context.clone();
+
+    // We used to clone `context` here to prevent assignments from applying to
+    // the caller. We now purposely preserve the caller's context with its
```

Comment:
```    // We used to clone `context` here to prevent assignments from applying to
    // the caller.```

Honestly i dont even think that was the original reason

we cloned just so we could set `in_call` for that scope and its children scopes, but we tried to avoid leaking `in_call = true` outside of this scope
https://github.com/posit-dev/ark/commit/17104159af642cd860fd0614765848e08cf68522

i think i was just copying some existing pattern being used in diagnostics, see other `context.clone()` calls in there

in retrospect your approach of setting it and making sure to reset the old value on the way out makes more sense in general, i think